scripts/install-nix-from-closure.sh

```diff
@@ -1,149 +1,618 @@
-    if [ $(($(sw_vers -productVersion | cut -d '.' -f 2))) -lt 10 ]; then
-        echo "$0: macOS $(sw_vers -productVersion) is not supported, upgrade to 10.10 or higher"
-        exit 1
+    if [ -z "$HOME" ]; then
+        error "
+
+    Environment variable HOME is not set.
+    "
     fi
 
-    printf '\e[1;31mSwitching to the Multi-User Darwin Installer\e[0m\n'
-    exec "$self/install-darwin-multi-user"
-    exit 0
-fi
+    # macOS support for 10.10 or higher
+    if [[ "$(uname -s)" = "Darwin" ]]; then
+        if [[ $(($(sw_vers -productVersion | cut -d '.' -f 2))) -lt 10 ]]; then
+            error "macOS $(sw_vers -productVersion) is not supported, upgrade to 10.10 or higher"
+        fi
+        notice 'Switching to the Multi-User Darwin Installer'
+        exec "$self/install-darwin-multi-user"
+        exit 0
+    fi
 
-if [ "$(id -u)" -eq 0 ]; then
-    printf '\e[1;31mwarning: installing Nix as root is not supported by this script!\e[0m\n'
-fi
+    if [ ! -e "$HOME" ]; then
+        error "
 
-echo "performing a single-user installation of Nix..." >&2
+    Home directory $HOME does not exist.
+    "
+    fi
+
+    if [ ! -d "$HOME" ]; then
+        error "
+
+    Home directory $HOME is not a directory.
+    "
+    fi
+
+    if [ ! -w "$HOME" ]; then
+        error "
+
+    Home directory $HOME is not writable for user $USER. No deal.
+    "
+    fi
+
+    if [ ! -O "$HOME" ]; then
+        error "
+
+    Home directory $HOME is not owned by user $USER,
+    users HOMEs probably must be owned by users.
+    "
+    fi
+
+    if [ ! -x "$HOME" ]; then
+        error "
+
+    Home directory $HOME is not marked as executable for user $USER,
+    how we are going to go into it?
+    "
+    fi
+
+    if [ -e "$dest" ]; then
+        # Destination directory exist. There probably Nix is/was installed before, be cautious.
+
+        if [ ! -d "$dest" ]; then
+            error "
+
+    '$dest' exists and it's not a directory.
+    "
+        fi
+
+        if [ ! -w $dest ]; then
+            error "
+
+    Directory '$dest' exists, but is not writable fot user '$USER'.
+
+    This could indicate that another user has already performed a single-user
+    installation of Nix on this system.
+
+    To enable multi-user support see
+    http://nixos.org/nix/manual/#ssec-multi-user.
+    To nevertheless do a single-user install for '$USER':
+    recursively set user '$USER' as owner for '$dest' directory.
+    "
+        fi
+
+    fi
+
+    if [ -e "$dest"/store ]; then
+
+        if [ ! -d "$dest"/store ]; then
+            error "
+
+    '$dest/store' exists and it's not a directory.
+    "
+        fi
+
+        if [ ! -w "$dest"/store ]; then
+            error "
+
+    Directory '$dest/store' exists, but is not writable fot user '$USER'.
+
+    This could indicate that another user has already performed a single-user
+    installation of Nix on this system.
+
+    If you wish to enable multi-user support see
```

Comment:
Since user can break system/of other user on the system, doing `chown` that we provided to him.
I decided to say in words, and make human to think what we does.
That `chown` cation is intuitive to people who know what they doing, but it is thought barrier for people who doesn't know what they doing, so they are going to research and understand what they are doing.

`contactme` is thrown from grahamc script, I not incorporated it really.

Reply:
Hmm ok, I suppose. Still, putting the word-explanation in a separate function would clean up the script a little bit.

Reply:
If there are just two sections that duplicate, I not add additional layer of abstraction.
Let two duplicate, reader will see that text duplicates, or will not see, - it doesn't matter for his understanding.
But he will not need to explore and put in head one more function.

If third like one appears (it is probable), - I will wrap.

Maybe wrap 'multi-user', that is logical. We can use it several times in script, even as disclaimer at the beginning, or the end.